protocol.go

```diff
@@ -47,10 +61,11 @@ func (p *Listener) Addr() net.Addr {
 
 // NewConn is used to wrap a net.Conn that may be speaking
 // the proxy protocol into a proxyproto.Conn
-func NewConn(conn net.Conn) *Conn {
+func NewConn(conn net.Conn, proxyHeaderPolicy Policy) *Conn {
```

Comment:
Rather than this change, I would suggest passing in something like a `opts *ConnOpts`. That way if you need to be able to pass more things in the future you don't have to break API again.

Reply:
I‘m not against such a change, but should it be really part of this PR? I think it would just inflate what this one is about and rather should be a separate one.

Reply:
You're changing the API already; I'm suggesting the API change be more forward looking. I think it would certainly be in-scope although I defer to @pires for preference.

Reply:
If we're changing the API already, I'd prefer to avoid changing it once more in a subsequent PR, particularly because this library is not tagged and folks most probably `go get` from master branch and there's the chance someone may get this functionality mid-flight and have the first version of the API instead of the final one - that I expect you to agree upon. Makes sense?

Reply:
Changed `NewConn()` with 001e22a to accept an arbitrary amount of options which can configure the connection.